test/optimize/test_optimize.py

```diff
@@ -259,8 +260,9 @@ def test_mpipoolengine():
              f'{path}/../../doc/example/example_MPIPool.py'])
 
         # read results
-        opt_result_reader = OptimizationResultHDF5Reader('temp_result.h5')
-        result1 = opt_result_reader.read()
+        result1 = read_result('temp_result.h5',
+                              problem=True,
+                              optimize=True)
```

Comment:
totally random remark: this is the style preferred in the python community. obviously no problem though

```suggestion
        result1 = read_result(
            'temp_result.h5',
            problem=True,
            optimize=True,
        )
```

Reply:
no that is good to hear as it is the style I like more. I actually thought this is the one the python community does not like as it creates more lines. thanks 🙏